kubernetes/resource_kubernetes_service_migrate.go

```diff
@@ -0,0 +1,202 @@
+						},
+						"selector": {
+							Type:        schema.TypeMap,
+							Description: "Route service traffic to pods with label keys and values matching this selector. Only applies to types `ClusterIP`, `NodePort`, and `LoadBalancer`. More info: http://kubernetes.io/docs/user-guide/services#overview",
+							Optional:    true,
+						},
+						"session_affinity": {
+							Type:        schema.TypeString,
+							Description: "Used to maintain session affinity. Supports `ClientIP` and `None`. Defaults to `None`. More info: http://kubernetes.io/docs/user-guide/services#virtual-ips-and-service-proxies",
+							Optional:    true,
+							Default:     "None",
+							ValidateFunc: validation.StringInSlice([]string{
+								"ClientIP",
+								"None",
+							}, false),
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Description: "Determines how the service is exposed. Defaults to `ClusterIP`. Valid options are `ExternalName`, `ClusterIP`, `NodePort`, and `LoadBalancer`. `ExternalName` maps to the specified `external_name`. More info: http://kubernetes.io/docs/user-guide/services#overview",
+							Optional:    true,
+							Default:     "ClusterIP",
+							ValidateFunc: validation.StringInSlice([]string{
+								"ClusterIP",
+								"ExternalName",
+								"NodePort",
+								"LoadBalancer",
+							}, false),
+						},
+						"health_check_node_port": {
+							Type:        schema.TypeInt,
+							Description: "Specifies the Healthcheck NodePort for the service. Only effects when type is set to `LoadBalancer` and external_traffic_policy is set to `Local`.",
+							Optional:    true,
+							Computed:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+			"wait_for_load_balancer": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Terraform will wait for the load balancer to have at least 1 endpoint before considering the resource created.",
+			},
+			"load_balancer_ingress": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"hostname": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesServiceStateUpgradeV0(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	log.Println("[INFO] Found Kubernetes Service state v0; upgrading state to v1")
+	upgradeLoadBalancerIngressV0(rawState)
+	delete(rawState, "load_balancer_ingress")
+	return rawState, nil
+}
+
+func upgradeLoadBalancerIngressV0(rawState map[string]interface{}) map[string]interface{} {
+	migratedIngressList := make([]map[string]interface{}, len(rawState), len(rawState))
```

Comment:
I'm planning to use this function for both Service and Ingress. I'll move it into `kubernetes/schema_resources_migrate.go` once the other StateUpgraders PR merges.